src/material/chips/testing/chip-harness-filters.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {BaseHarnessFilters} from '@angular/cdk/testing';
+
+/** A set of criteria that can be used to filter a list of `MatChipHarness` instances. */
+export interface ChipHarnessFilters extends BaseHarnessFilters {
+  /** Only find instances whose text matches the given value. */
+  text?: string | RegExp;
+  /** Only find chip instances whose selected state matches the given value. */
+  selected?: boolean;
```

Comment:
The mdc versions of chips has types of chips that don't have a `selected` state- this would affect the filter as well as the chip harness API. Would it make sense to author the harness for the existing chip-list against the concepts of the MDC-based chips so that we're going in the direction we want? (e.g `MatChipListbox`, `MatChipOption`, etc.)

Reply:
Wouldn't those just always return `false`? I think that no matter what, we'll have some differences in the harnesses between the current ones and MDC, because the public API is different.

Reply:
IIRC the MDC-based `mat-chip-listbox` should be equivalent to the current `mat-chip-list`, so maybe the end result should look something like

`MatChipHarness` - any kind of chip, no selection API
`MatChipSetHarness` - any kind of chip container, no selection API

`MatChipOptionHarness` - specifically a listbox (option) chip, extends `MatChipHarness`, selection API
`MatChipListboxHarness` - specifically listbox chip container, extends `MatChipSetHarness`, selection API
`MatChipListHarness` becomes an alias for `MatChipListboxHarness`

Thoughts?

Reply:
IMO that makes sense for the MDC chips, but not for the existing ones. Somebody that hasn't used the MDC chips won't know what a "chip listbox" is referring to. I don't think it's worth it to avoid breaking changes in the harnesses, considering that the component's public API has breaking changes already.

Reply:
Yeah I think since one of the main goals for the harnesses is to help with the MDC migration, we should align the API with the current version of chips for now. Once the migration is done we may want to change it to align better with the new API of the MDC version.

Reply:
>  Once the migration is done

The existing API will be kicking around for a while yet, though. Anything we can do now to align them more closely will make less work down the road. The container part is pretty straightforward- `MatChipList` becomes an alias for `MatChipListbox` (the latter also not needing to be exported presently). The chip part is the problem- if we add the selection APIs to `MatChipHarness`, we're locking ourselves into every type of chip having a selection API when we know that won't be the case going forward. We could just make them no-ops / always return false, but that feels wrong since the presence of those APIs signals to a user that those chips _can_ be selected.

Reply:
How would this all work with the current version of chips though? Would there would be 2 different harnesses for `MatChip` (`MatChipHarness`, `MatChipOptionHarness`) and users just pick the ones that gives the functionality they want in their test?

Reply:
It's worth noting that `selectable` so a chip could change type between change detection runs. That means that people might have to use two harnesses for the same element.

Reply:
I think you could just use `MatOptionHarness` exclusively and it would report `false` if it wasn't selectable, so I don't think you'd need to use both for a single element (unless `MatChipHarness` does something that `MatChipOptionHarness` doesn't)

Reply:
That's the point I was making above about always using `MatChipOption`. Anyway, I don't think that we can get away with using the exact same set of harnesses for both MDC and the current version, because they're fundamentally different APIs.

Reply:
I suppose I had just been imagining that we could just completely omit the selection APIs from `MatChipHarness` for now, and only add them to the mdc ones. 

I admit it's not ideal. Thinking about it more, I can live with going with this API for now and just committing to writing migration schematics for chips specifically because the APIs should be pretty easy to transform.